packages/ontario-design-system-component-library/src/components/ontario-task/ontario-task.tsx

```diff
@@ -0,0 +1,221 @@
+	 *
+	 * This is used for translations, and is by default set through event listeners checking for a language property from the header. If no language is passed, it will default to English.
+	 */
+	@Prop({ mutable: true }) language?: Language;
+
+	/**
+	 * Used to include the ontario-hint-text component for the task.
+	 *
+	 * This is optional.
+	 */
+	@Prop() hintText?: string | Hint;
+
+	/**
+	 * Defines the status of the task, with default set to 'NotStarted'.
+	 *
+	 * Accepts values from `TaskStatuses` enum: `NotStarted`, `InProgress`, `Completed`.
+	 */
+	@Prop() taskStatus: TaskStatus;
+
+	/**
+	 * Used for the `aria-describedby` value of the task's label. This will match with the id of the hint text.
+	 */
+	@State() hintTextId: string | null | undefined;
+
+	/**
+	 * The hint text options are re-assigned to the internalHintText array.
+	 */
+	@State() private internalHintText: Hint;
+
+	/**
+	 * Mutable variable, for internal use only.
+	 *
+	 * Set the task's status state depending on validation result.
+	 */
+	@State() private taskStatusState: TaskStatus;
+
+	/**
+	 * Watch for changes in `taskStatus` prop to validate its value.
+	 */
+	@Watch('taskStatus')
+	validateTaskStatus() {
+		const isValidStatus = validateValueAgainstArray(this.taskStatus, TaskStatuses);
+		this.taskStatusState = isValidStatus ? this.taskStatus : this.warnDefaultTaskStatus();
+	}
+
+	/**
+	 * Display a console warning if `taskStatus` is invalid and set to a default value.
+	 */
+	private warnDefaultTaskStatus(): TaskStatus {
+		const message = new ConsoleMessageClass();
+		message
+			.addDesignSystemTag()
+			.addMonospaceText(' taskStatus ')
+			.addRegularText('on')
+			.addMonospaceText(' <ontario-task> ')
+			.addRegularText('was set to an invalid taskStatus; only ')
+			.addMonospaceText(TaskStatuses.join(', '))
+			.addRegularText(' are supported. The default taskStatus ')
+			.addMonospaceText('notStarted')
+			.addRegularText(' is assumed.')
+			.printMessage();
+		return 'notStarted';
+	}
+
+	/**
+	 * Watch for changes in `hintText` prop and parse it if available.
+	 */
+	@Watch('hintText')
+	private parseHintText() {
+		if (this.hintText) {
+			const hintTextObject = constructHintTextObject(this.hintText);
+			this.internalHintText = hintTextObject;
+		}
+	}
+
+	/**
+	 * Listen for app language settings on the window to update the component language.
+	 */
+	@Listen('setAppLanguage', { target: 'window' })
+	handleSetAppLanguage(event: CustomEvent<Language>) {
+		if (!this.language) {
+			this.language = validateLanguage(event);
+		}
+	}
+
+	/**
+	 * Listen for header language toggling events to update the language dynamically.
+	 */
+	@Listen('headerLanguageToggled', { target: 'window' })
+	handleHeaderLanguageToggled(event: CustomEvent<Language>) {
+		this.language = validateLanguage(event);
+	}
+
+	/**
+	 * Determines the badge color based on the current `taskStatusState`.
+	 */
+	private getBadgeColor(): 'black' | 'light-teal' | 'grey' {
```

Comment:
```suggestion
	private getBadgeColour(): 'black' | 'light-teal' | 'grey' {
```

As well, we have a TaskBadgeColour type that can be used as the return type here.